test/graphtile.cc

```diff
@@ -16,6 +16,9 @@ struct testable_graphtile : public valhalla::baldr::GraphTile {
     header_->set_edge_bin_offsets(offsets);
     edge_bins_ = bins.data();
   }
+  ~testable_graphtile() {
+    delete header_;
```

Comment:
so a couple lines above we make a new tile header just to do the test and we were never deleting it. seemed not needed for the test because the program would end anyway but this shuts up the ASAN lib

Reply:
I suppose this is OK for the test, but wouldn't it be better practice to use a `std::unique_ptr` member variable on the struct and let RAII take care of the cleanup?

Reply:
seemed like more trouble than just doing `delete`. to make that work id have to add another member variable just so i can point the inherited member at it. would you prefer that? to me it seemed more circuitus